src/cli/run_command.zig

```diff
@@ -227,7 +234,387 @@ pub const RunCommand = struct {
+            pid_fd: bun.FileDescriptor,
+        ) !?*LifecycleScriptSubprocess {
+            // TODO: this doesnt handle some cleanup edge cases on error
+            var this = try manager.allocator.create(LifecycleScriptSubprocess);
+            errdefer this.deinit(manager.allocator);
+
+            this.* = .{
+                .package_name = package_name,
+                .script_name = script_name,
+                .manager = manager,
+                .scripts = scripts,
+                .current_script_index = script_index,
+                .waitpid_result = null,
+                .output_buffer = .{},
+                .pid_poll = Async.FilePoll.initWithPackageManager(
+                    manager,
+                    pid_fd,
+                    .{},
+                    @as(*PidPollData, @ptrCast(this)),
+                ),
+                .stdout_poll = Async.FilePoll.initWithPackageManager(manager, stdout_fd, .{}, this),
+                .stderr_poll = Async.FilePoll.initWithPackageManager(manager, stderr_fd, .{}, this),
+            };
+
+            try this.registerPolls();
+
+            return this;
+        }
+
+        pub fn registerPolls(this: *LifecycleScriptSubprocess) !void {
+            _ = try this.stdout_poll.register(this.manager.uws_event_loop, .readable, false).unwrap();
+            _ = try this.stderr_poll.register(this.manager.uws_event_loop, .readable, false).unwrap();
+
+            switch (this.pid_poll.register(
+                this.manager.uws_event_loop,
+                .process,
+                true,
+            )) {
+                .result => {},
+                .err => |err| {
+                    // Sometimes the pid poll can fail to register if the process exits
+                    // between posix_spawn() and pid_poll.register(), but it is unlikely.
+                    // Any other error is unexpected here.
+                    if (err.getErrno() != .SRCH) {
+                        @panic("This shouldn't happen. Could not register pid poll");
+                    }
+
+                    this.onProcessUpdate(0);
+                },
+            }
+        }
+
+        pub fn onOutputUpdate(this: *LifecycleScriptSubprocess, size: i64, fd: bun.FileDescriptor) void {
+            // var needed_capacity = this.output_buffer.len + @as(u32, @intCast(size));
+            // _ = needed_capacity;
+            this.output_buffer.ensureUnusedCapacity(this.manager.allocator, @intCast(size)) catch @panic("Failed to allocate memory for output buffer");
+
+            if (size == 0) {
+                this.finished_fds +|= 1;
+                if (this.waitpid_result) |result| {
+                    if (this.finished_fds == 2) {
+                        this.onResult(result);
+                    }
+                }
+                return;
+            }
+
+            var remaining = size;
+            while (remaining > 0) {
+                const n: u32 = @truncate(std.os.read(
```

Comment:
Let's avoid using std.os.read here. `std.os.read` uses `unreachable` which is unsafe. Instead, we can use `bun.sys.read` and that will also give us better quality error messages